plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultQueryBuilder.java

```diff
@@ -174,6 +176,44 @@ public PreparedQuery prepareDeleteQuery(
         return new PreparedQuery(sql, accumulator.build());
     }
 
+    @Override
+    public PreparedQuery prepareUpdateQuery(
+            JdbcClient client,
+            ConnectorSession session,
+            Connection connection,
+            JdbcNamedRelationHandle baseRelation,
+            TupleDomain<ColumnHandle> tupleDomain,
+            Optional<ParameterizedExpression> additionalPredicate,
+            Map<String, Domain> assignments,
+            List<JdbcColumnHandle> columns)
+    {
+        Map<String, JdbcColumnHandle> columnMap = columns.stream().collect(Collectors.toMap(JdbcColumnHandle::getColumnName, Function.identity()));
+        ImmutableList.Builder<QueryParameter> accumulator = ImmutableList.builder();
+
+        String sql = "UPDATE " + getRelation(client, baseRelation.getRemoteTableName()) + " SET ";
+
+        sql += assignments.entrySet().stream().map(entry -> {
```

Comment:
This code is hard to read, can we extract some method out of it?

Reply:
refactored to be more readable